static/app/components/pagination.tsx

```diff
@@ -0,0 +1,98 @@
+import {browserHistory, withRouter, WithRouterProps} from 'react-router';
+import styled from '@emotion/styled';
+import {Query} from 'history';
+
+import Button from 'app/components/button';
+import ButtonBar from 'app/components/buttonBar';
+import {IconChevron} from 'app/icons';
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import parseLinkHeader from 'app/utils/parseLinkHeader';
+
+/**
+ * @param cursor The string cursor value
+ * @param path   The current page pathname
+ * @param query  The current query object
+ * @param delta  The delta in page number change triggered by the
+ *               click. A negative delta would be a "previous" page.
+ */
+export type CursorHandler = (
+  cursor: string | undefined,
+  path: string,
+  query: Query,
+  delta: number
+) => void;
+
+type Props = WithRouterProps & {
+  pageLinks?: string | null;
+  to?: string;
+  caption?: React.ReactElement;
+  size?: 'zero' | 'xsmall' | 'small';
+  onCursor?: CursorHandler;
+  disabled?: boolean;
+  className?: string;
+};
+
+const defaultOnCursor: CursorHandler = (cursor, path, query, _direction) =>
+  browserHistory.push({
+    pathname: path,
+    query: {...query, cursor},
+  });
+
+const Pagination = ({
+  to,
+  location,
+  className,
+  onCursor = defaultOnCursor,
+  pageLinks,
+  size = 'small',
+  caption,
+  disabled = false,
+}: Props) => {
+  if (!pageLinks) {
+    return null;
+  }
+
+  const path = to ?? location.pathname;
+  const query = location.query;
+  const links = parseLinkHeader(pageLinks);
+  const previousDisabled = disabled || links.previous.results === false;
+  const nextDisabled = disabled || links.next.results === false;
```

Comment:
```suggestion
  const previousDisabled = disabled || links.previous?.results === false;
  const nextDisabled = disabled || links.next?.results === false;
```

Could be more undefined here.

Reply:
Thanks for catching that.